tests/zip.rs

```diff
@@ -42,6 +43,20 @@ fn test_double_ended_zip_longest() {
     assert_eq!(it.next(), None);
 }
 
+#[test]
+fn test_double_ended_zip() {
```

Comment:
Could you formulate this as a `quickcheck` test?

I'd like to see that for any two inputs, `multizip((a, b)).collect_vec().rev()` is equal to (something like) `multizip((a, b)).rfold(Vec::new(), |vec, e| { vec.push(e); vec })`